plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+            fullcmd = fullcmd + ' WAIT'
+            parmtext = parmtext + '\n      WAIT'
+
+        if 'NOSECURITY' in opts or 'nosecurity' in opts:
+            fullcmd = fullcmd + ' NOSECURITY'
+            parmtext = parmtext + '\n      NOSECURITY'
+        else:
+            fullcmd = fullcmd + ' SECURITY'
+            parmtext = parmtext + '\n      SECURITY'
+
+        if len(sysname) > 1:
+            fullcmd = "{0} SYSNAME\\({1}\\)".format(fullcmd, sysname)
+            parmtext = "{0}\n      SYSNAME({1})".format(parmtext, sysname)
+
+        if len(automove) > 1:
+            fullcmd = fullcmd + ' ' + automove
+            parmtext = parmtext + '\n      ' + automove
+            if(len(automove_list) > 1):
+                fullcmd = fullcmd + '(' + automove_list + ')'
+                parmtext = parmtext + '(' + automove_list + ')'
+        parmtext = parmtext + parmtail
+
+    if gonna_unmount:     # unmount/remount
+        fullumcmd = "tsocmd UNMOUNT FILESYSTEM\\(\\'{0}\\'\\)".format(src)
+        if len(unmount_opts) < 2:
+            unmount_opts = "NORMAL"
+            fullumcmd = fullcmd + ' ' + unmount_opts
+
+    comment = ''
+
+    if gonna_unmount:
+        if currently_mounted:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullumcmd, use_unsafe_shell=True)
+                    comment += "ran unmount command\n"
+                    currently_mounted = False
+                except Exception as err:
+                    module.fail_json(msg=str(err), stderr=str(res_args))
+            else:
+                comment += "(unmount) NO Action taken: ANSIBLE CHECK MODE\n"
+                stdout = 'ANSIBLE CHECK MODE'
+        else:
+            comment += "Unmount called on data set that is not mounted.\n"
+
+    if gonna_mount:
+        if currently_mounted is False:
+            changed = True
+            if module.check_mode is False:
+                try:
+                    (rc, stdout, stderr) = module.run_command(
+                        fullcmd, use_unsafe_shell=True)
+                    comment = "ran command\n"
+                except Exception as err:
+                    module.fail_json(msg=str(err), stderr=str(res_args))
+            else:
+                comment += "(mount) NO Action taken: ANSIBLE CHECK MODE\n"
+                stdout = 'ANSIBLE CHECK MODE'
+        else:
+            comment += "Mount called on data set that is already mounted.\n"
+
+    rc = 0
+    stdout = stderr = None
+
+    if write_fstab and module.check_mode is False:
+        fst_du = data_set.DataSetUtils(fstab)
+        fst_exists = fst_du.exists()
+        if fst_exists is False:
+            module.fail_json(
+                msg="fstab set member (" + fstab + ") doesn't exist",
+                stderr=str(res_args)
+            )
+
+        tmp_file = tempfile.NamedTemporaryFile(delete=True)
```

Comment:
is this temp file being used, or is it deleted before usage?

Reply:
Yes to both questions.  I'm using this is to generate a temp file filename, which is closed & deleted before use (so we're not opening an unknown file).